app/src/main/java/org/simple/clinic/bloodsugar/BloodSugarReading.kt

```diff
@@ -33,16 +35,32 @@ data class BloodSugarReading(val value: String, val type: BloodSugarMeasurementT
       }
     }
 
-  val displayValue: String
-    get() {
-      return when (type) {
-        Random -> value.toFloat().toInt().toString()
-        PostPrandial -> value.toFloat().toInt().toString()
-        Fasting -> value.toFloat().toInt().toString()
-        HbA1c -> value.toFloat().toString()
-        is Unknown -> value.toInt().toString()
-      }
+  fun displayValue(bloodSugarUnitPreference: BloodSugarUnitPreference): String {
+    return when (bloodSugarUnitPreference) {
+      Mg -> displayValueMg()
+      Mmol -> displayValueMmol()
+    }
+  }
+
+  private fun displayValueMg(): String {
+    return when (type) {
+      Random -> value.toFloat().toInt().toString()
+      PostPrandial -> value.toFloat().toInt().toString()
+      Fasting -> value.toFloat().toInt().toString()
+      HbA1c -> value.toFloat().toString()
+      is Unknown -> value.toInt().toString()
     }
+  }
+
+  private fun displayValueMmol(): String {
```

Comment:
Did we miss doing the conversion from mg/dl to mmol/l here?

Reply:
OMG yes, Since we are no longer using `toMmol` we should be doing that here. Fixing it 😅